File: _Collection_of_Problems/Backjoon/11723/main.py
import sys

input = sys.stdin.readline

def solve():
    try:
        M = int(input())
    except (ValueError, IndexError):
        return

    S = 0
    
    for _ in range(M):
        command_line = input().rstrip().split()
        cmd = command_line[0]

        if cmd == "add":
            x = int(command_line[1])
            S |= (1 << x)
        elif cmd == "remove":
            x = int(command_line[1])
            S &= ~(1 << x)
        elif cmd == "check":
            x = int(command_line[1])
            if S & (1 << x):
                print("1")
            else:
                print("0")
        elif cmd == "toggle":
            x = int(command_line[1])
            S ^= (1 << x)
        elif cmd == "all":
            S = (1 << 21) - 1
        elif cmd == "empty":
            S = 0

if __name__ == "__main__":
    solve()

File: _Collection_of_Problems/Backjoon/11723/Main.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
import java.io.IOException;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int S = 0;
        int M = Integer.parseInt(br.readLine());
        for(int i=0; i<M; i++){
            StringTokenizer st = new StringTokenizer(br.readLine());
            String cmd = st.nextToken();
            int x;
            
            switch(cmd){
                case "add":
                    x = Integer.parseInt(st.nextToken());
                    S |= (1 << x);
                    break;
                case "remove":
                    x = Integer.parseInt(st.nextToken());
                    S &= ~(1 << x);
                    break;
                case "check":
                    x = Integer.parseInt(st.nextToken());
                    sb.append((S & (1 << x)) == 0 ? 0 : 1).append("\n");
                    break;
                case "toggle":
                    x = Integer.parseInt(st.nextToken());
                    S ^= (1 << x);
                    break;
                case "all":
                    S = (1 << 21) - 1;
                    break;
                case "empty":
                    S = 0;
                    break;
            }
        }
        System.out.println(sb);        
    }
}